rotkehlchen/db/search_assets.py

```diff
@@ -0,0 +1,134 @@
+from typing import TYPE_CHECKING, Any, Dict, List, Optional, Tuple
+
+from polyleven import levenshtein
+
+from rotkehlchen.assets.types import AssetType
+from rotkehlchen.constants.assets import A_ETH, A_ETH2
+from rotkehlchen.constants.resolver import ChainID
+from rotkehlchen.globaldb.handler import ALL_ASSETS_TABLES_QUERY, GlobalDBHandler
+
+if TYPE_CHECKING:
+    from rotkehlchen.db.dbhandler import DBHandler
+    from rotkehlchen.db.drivers.gevent import DBCursor
+
+
+def _search_only_nfts_levenstein(
+        cursor: 'DBCursor',
+        substring_search: str,
+) -> List[Tuple[int, Dict[str, Any]]]:
+    cursor.execute(
+        'SELECT identifier, name, collection_name FROM nfts '
+        'WHERE name LIKE ? OR collection_name LIKE ? ',
+        (f'%{substring_search}%', f'%{substring_search}%'),
+    )
+    search_result: List[Tuple[int, Dict[str, Any]]] = []
+    for entry in cursor:
+        # default to a value greater than the threshold to avoid false-positives
+        # and prevent raising `UnboundLocalError` when name/collection name is None.
+        lev_dist_min = 6
+        if entry[1] is not None:
+            lev_dist_min = min(
+                lev_dist_min,
+                levenshtein(substring_search, entry[1].casefold()),
+            )
+        if entry[2] is not None:
+            lev_dist_min = min(
+                lev_dist_min,
+                levenshtein(substring_search, entry[2].casefold()),
+            )
+        entry_info = {
+            'identifier': entry[0],
+            'name': entry[1],
+            'collection_name': entry[2],
+            'asset_type': AssetType.NFT.serialize(),
+        }
+        search_result.append((lev_dist_min, entry_info))
+
+    return search_result
+
+
+def _search_only_assets_levenstein(
+        cursor: 'DBCursor',
+        db: 'DBHandler',
+        substring_search: str,
+        evm_chain: Optional[ChainID],
+) -> List[Tuple[int, Dict[str, Any]]]:
+    search_result: List[Tuple[int, Dict[str, Any]]] = []
+    resolved_eth = A_ETH.resolve_to_crypto_asset()
+    # Build query manually since we cannot use db/filtering.py due to different structures of
```

Comment:
But you have 2 different functions and this is the assets function. Why can't you use the assets filter here?